app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people.update({self.name: self})
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    person_list = [Person(person["name"], person["age"]) for person in people]
+
+    for person in people:
+        if person.get("wife"):
+            (Person.people[person.get("name")].__dict__
+             .update({"wife": Person.people[person.get("wife")]}))
+
+        if person.get("husband"):
+            (Person.people[person.get("name")].__dict__
+             .update({"husband": Person.people[person.get("husband")]}))
```

Comment:
Try to simplify this code. We don't need to use `update` method for this.

Example:
`Person.people[name].wife = Person.people[wife_name]`